specs/merge/beacon-chain.md

```diff
@@ -196,12 +198,15 @@ def process_block(state: BeaconState, block: BeaconBlock) -> None:
 
 #### `process_execution_payload`
 
+*Note:* This function depends on `process_randao` function call as it retrieves the most recent randao mix from the `state`. Implementations that are considering parallel processing of execution payload with respect to beacon chain state transition function should work around this dependency.
```

Comment:
I would lean towards removing this note. It's "obvious" that there is a parallelisation opportunity and client implementers have become experts at handling optimisations that the spec doesn't address.

Reply:
IMO, the second statement may be removed as it can be deduced from the first one. Don't you think that removing this entire comment would affect readability? How common the spec is in leaving implicit dependencies unattended?

Reply:
I think it's pretty reasonable as is, plus I'm cutting a release today and want to get this in.

Let's leave it here for now. Remove if this info makes it into somewhere else -- e.g. annotated spec or implementaton guide or something